### ISSUE TITLE
GroupBy() followed by Count() tries to perform count on client

### ISSUE THREAD

**Issue by MaceWindu** [linq2db/linq2db#781]
At least following linq patterns generate `select *` query (which fails, because usually not all table fields included into group by) instead of `select count(*):
```c#
query.GroupBy(...).Count();
query.GroupBy(...).LongCount();
query.GroupBy(...).Having(...).Count();
query.GroupBy(...).Having(...).LongCount();
```

### Environment details
linq2db version: *1.8.3*  
Database Server: *SQL Server 2016*  
Operating system: *Windows 10*  
Framework version: *.NET Framework 4.6.1*

**Comment by MaceWindu** [comment]
BTW, for workaround, add `Select(...)` call before `Count`

**Comment by MaceWindu** [comment]
Looks like this happens only if you group not by plain field, but by some expression. Will add tests soon

**Comment by ili** [comment]
fixed via #775

**Comment by Preservejohan3** [comment]
> BTW, for workaround, add `Select(...)` call before `Count`





https://eddyencouragement.github.io/return-value36/
